File: trunk/PolarMathCS/PolarMath/Evaluation/HRV/BP.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PolarMath.Evaluation.HRV
{
    public class BP : IEvaluation<double>
    {
        private const int LowBorder = 400;
        private const int HighBorder = 1300;

        public Double Evaluate(Training training) {
		    var intervals = training.Intervals;
		    var localIntervals = intervals.Where(interval => interval >= LowBorder && interval <= HighBorder).ToList();


		    var maxInt = localIntervals.Max();
		    var minInt = localIntervals.Min();
		
		    return (maxInt - minInt) / (double) 1000;
	    }
    }
}

File: PolarMath/src/org/macondo/polar/evaluation/hrv/BP.java
package org.macondo.polar.evaluation.hrv;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;

public class BP implements Evaluation<Double> {
	private static final int lowBorder = 400;
	private static final int highBorder = 1300;
	
	public Double evaluate(Training training) {
		final List<Integer> intervals = training.getIntervals();
		List<Integer> localIntervals = new ArrayList<Integer>(intervals);
		
		List<Integer> intervalsToRemove = new ArrayList<Integer>();
		for (int interval : localIntervals) {
			if (interval < lowBorder || interval > highBorder) {
				intervalsToRemove.add(interval);
			}
		}
		
		localIntervals.removeAll(intervalsToRemove);
		int maxInt = Collections.max(localIntervals);
		int minInt = Collections.min(localIntervals);		
		
		return (maxInt - minInt) / (double) 1000;
	}

}
